src/state.py

```diff
@@ -11,6 +12,62 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        #if traded the subject as expected get score
+        for i in self.students.values(): 
+
+            #checking buddies 
+            for numbers in set(list(i.buddies.values())):
+                    for classes in set(list(i.buddies.keys())): 
+                        if i.subjects_and_classes[classes] == self.students[numbers].subjects_and_classes[classes]: 
+                            score += 30         #the person is at the same class of his friend
+                            i.alone = False     #Flag-- 
+                        else: 
+                            score -= 20
+                        
+            #checking target class
+            for position, j in enumerate(i.subjects_and_classes): 
+                if j in i.subject_targets.keys(): 
+                    if i.subjects_and_classes[j] in i.subject_targets[j]:  
+                        score += 40         #if the person got the target class 
+                        i.target = True     #Flag--
+                    else: 
+                        score -= 30
+
+                #checking givin classes 
+                if j in i.subject_give_ins.keys():
+                    if i.subjects_and_classes[j] in i.subject_give_ins[j]: 
+                        score -= 3          #if a class was abdicated
+                        i.givin = True    #Flag--
+                    else: 
+                        score += 5
+                     
+
+               #checking schedule conflicts. If two classes are incompatible then score = -inf 
+                for p in range(position+1, len(list(i.subjects_and_classes.keys()))):
+                    key = list(i.subjects_and_classes.keys())[p] 
+                    sched_1 = self.class_schedules[translate_subject_and_class(j,i.subjects_and_classes[j])]
+                    sched_2 = self.class_schedules[translate_subject_and_class(key, i.subjects_and_classes[key])] 
+                    if (sched_1.conflicts(sched_2)):
+                        score = float('-inf')
+                        break 
+
+            # Reading the method of gen states we have that it's just possible to trade classes in the givin list 
+            # and target list, but the classes can remain the same. 
+            # If there's no change in the classses, then givin = False and target = False 
+
+            # If he gives up a class but didn't get the target nor he is with his buddy, then infinity 
+            if i.givin and i.target and i.alone and list(i.buddies): 
+                score = float('-inf')
+                break 
+            # If he gives up a class but didn't get the target and did not choose a buddy 
+            if i.givin and not i.target and not list(i.buddies):
+                score = float('-inf')
+                break 
+
```

Comment:
Conditions changed. Explanation: 

> Since the gen state just make trades between the actual classes, givin classes, and target classes, we have that if the student didn't give up any classes and didn't get any target it means that he is with the same class as before. So the score is no -inf. 
> If he gives up a class and is not with his buddy neither a target class, it should be -inf. 
> If he gives up class, didn't get a target class and didn't choose a buddy, then the score must be -inf. 

I don't if I forgot something

Reply:
I'm sorry, but I'm having a hard time understanding this @Jumaruba 
Do you mean that the original state has a score of -inf,  because each student current class at a a given subject is different than his targets and his give ins? 
If so, yes, we should address this. 
I'll take care of it, using this branch.